File: software/examples/php/ExampleFindBorders.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletJoystick.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletJoystick;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Joystick Bricklet

// Callback function for position reached callback
function cb_positionReached($x, $y)
{
    if ($y == 100) {
        echo "Top\n";
    } elseif ($y == -100) {
        echo "Bottom\n";
    }

    if ($x == 100) {
        echo "Right\n";
    } elseif ($x == -100) {
        echo "Left\n";
    }

    echo "\n";
}

$ipcon = new IPConnection(); // Create IP connection
$j = new BrickletJoystick(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get threshold callbacks with a debounce time of 0.2 seconds (200ms)
$j->setDebouncePeriod(200);

// Register position reached callback to function cb_positionReached
$j->registerCallback(BrickletJoystick::CALLBACK_POSITION_REACHED, 'cb_positionReached');

// Configure threshold for position "outside of -99, -99 to 99, 99"
$j->setPositionCallbackThreshold('o', -99, 99, -99, 99);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/c/example_find_borders.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_joystick.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change XYZ to the UID of your Joystick Bricklet

// Callback function for position reached callback
void cb_position_reached(int16_t x, int16_t y, void *user_data) {
	(void)user_data; // avoid unused parameter warning

	if(y == 100) {
		printf("Top\n");
	} else if(y == -100) {
		printf("Bottom\n");
	}

	if(x == 100) {
		printf("Right\n");
	} else if(x == -100) {
		printf("Left\n");
	}

	printf("\n");
}

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	Joystick j;
	joystick_create(&j, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Get threshold callbacks with a debounce time of 0.2 seconds (200ms)
	joystick_set_debounce_period(&j, 200);

	// Register position reached callback to function cb_position_reached
	joystick_register_callback(&j,
	                           JOYSTICK_CALLBACK_POSITION_REACHED,
	                           (void (*)(void))cb_position_reached,
	                           NULL);

	// Configure threshold for position "outside of -99, -99 to 99, 99"
	joystick_set_position_callback_threshold(&j, 'o', -99, 99, -99, 99);

	printf("Press key to exit\n");
	getchar();
	joystick_destroy(&j);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}
